resources.whatwg.org/build/deploy.sh

```diff
@@ -76,46 +63,47 @@ run_post_build_step() {
     fi
 }
 
-if [[ "$BRANCH" == "master" ]] ; then
-    # Commit snapshot, if master
-    COMMIT_DIR="$WEB_ROOT/$COMMITS_DIR/$SHA"
-    mkdir -p "$COMMIT_DIR"
-    curl https://api.csswg.org/bikeshed/ -f -F file=@"$INPUT_FILE" -F md-status=LS-COMMIT \
-         -F md-warning="Commit $SHA $COMMIT_URL_BASE$SHA replaced by $LS_URL" \
-         -F md-title="$TITLE (Commit Snapshot $SHA)" \
-         -F md-Text-Macro="SNAPSHOT-LINK $BACK_TO_LS_LINK" \
-         > "$COMMIT_DIR/index.html";
-    copy_extra_files "$COMMIT_DIR"
-    run_post_build_step "$COMMIT_DIR"
-    echo "Commit snapshot output to $COMMIT_DIR"
-    echo ""
+# Commit snapshot
+COMMIT_DIR="$WEB_ROOT/$COMMITS_DIR/$SHA"
+mkdir -p "$COMMIT_DIR"
+curl https://api.csswg.org/bikeshed/ -f -F file=@"$INPUT_FILE" -F md-status=LS-COMMIT \
+     -F md-warning="Commit $SHA $COMMIT_URL_BASE$SHA replaced by $LS_URL" \
+     -F md-title="$TITLE (Commit Snapshot $SHA)" \
+     -F md-Text-Macro="SNAPSHOT-LINK $BACK_TO_LS_LINK" \
+     > "$COMMIT_DIR/index.html";
+copy_extra_files "$COMMIT_DIR"
+run_post_build_step "$COMMIT_DIR"
+echo "Commit snapshot output to $COMMIT_DIR"
+echo ""
 
-    # Living standard, if master
-    curl https://api.csswg.org/bikeshed/ -f -F file=@"$INPUT_FILE" \
-         -F md-Text-Macro="SNAPSHOT-LINK $SNAPSHOT_LINK" \
-         > "$WEB_ROOT/index.html"
-    copy_extra_files "$WEB_ROOT"
-    run_post_build_step "$WEB_ROOT"
+# Living standard
+curl https://api.csswg.org/bikeshed/ -f -F file=@"$INPUT_FILE" \
+     -F md-Text-Macro="SNAPSHOT-LINK $SNAPSHOT_LINK" \
+     > "$WEB_ROOT/index.html"
+copy_extra_files "$WEB_ROOT"
+run_post_build_step "$WEB_ROOT"
+echo "Living standard output to $WEB_ROOT"
+echo ""
 
-    SERVICE_WORKER_SHA=$(curl --fail https://resources.whatwg.org/standard-service-worker.js | shasum | cut -c 1-40)
-    echo "\"use strict\";
+# Standard service worker
+SERVICE_WORKER_SHA=$(curl --fail https://resources.whatwg.org/standard-service-worker.js | shasum | cut -c 1-40)
+echo "\"use strict\";
 importScripts(\"https://resources.whatwg.org/standard-service-worker.js\");
 // Version (for service worker freshness check): $SERVICE_WORKER_SHA" \
-       > "$WEB_ROOT/service-worker.js"
-
-    echo "Living standard output to $WEB_ROOT"
-    echo ""
+     > "$WEB_ROOT/service-worker.js"
+echo "Service worker script output to $WEB_ROOT"
+echo ""
 
-    find "$WEB_ROOT" -type f -print
-    echo ""
-fi
+find "$WEB_ROOT" -type f -print
+echo ""
 
+# Run the HTML checker only when building on Travis
 if [[ "$TRAVIS" == "true" ]]; then
-    # Run the HTML checker only when building on Travis
     curl -O https://sideshowbarker.net/nightlies/jar/vnu.jar
     /usr/lib/jvm/java-8-oracle/jre/bin/java -jar vnu.jar --skip-non-html --Werror --filterpattern "$CHECKER_FILTER" "$WEB_ROOT"
 fi
 
+# Deploy from Travis on master branch only
 if [[ "$TRAVIS" == "true" && "$BRANCH" == "master" ]]; then
```

Comment:
I think this will attempt to deploy when building a pull request, as long as the PR is from the fork's master branch. (Because you deleted the guard against pull requests.)

Reply:
After a bit of flailing about I confirmed that you're right in https://travis-ci.org/whatwg/encoding/builds/318824786, TRAVIS_BRANCH will be master even for a PR. (I thought it might be some generated name like pr/123/merged/with/master, but nope.)

Will fix.

Reply:
Actually I was still a bit wrong, the name of the source branch doesn't seem to matter, https://docs.travis-ci.com/user/environment-variables/#Default-Environment-Variables says "for builds triggered by a pull request this is the name of the branch targeted by the pull request".